### DISCUSSION TITLE
Automatic unpacking .nzb's stored in zip/rar/7z files in the incoming dir

### DISCUSSION THREAD

**Discussion by bacon-cheeseburger** [nzbgetcom/nzbget#191] [General]
Hi,
I used to have an extension that would automatically monitor the incoming dir and unpack any zip/rar/7z files that contained .nzb's so they could be picked up in the queue. I lost the extension however, does anyone happen to have one? Was also thinking this might be a good extension to have in the official extension repo. Or, maybe it could be added to the existing ExtendedUnrar extension?

**Comment by luckedea** [comment]
@bacon-cheeseburger This should be trivial to implement, but I'm curious about your use pattern, can you tell more why this is needed? I don't seem to have one or find one, going to search more. If someone shares the actual extension - of course this would be preferred, we'll just adapt it for fresh python and add a test to it. Do you recall the name?

**Reply by bacon-cheeseburger** [reply]
IIRC it was just call "unzip". Some sources allow you to select multiple different nzb's and download them all as a single zip/7z file instead of each file individually. It's also an easy way to bundle entire seasons or series. I guess I'd consider it a quality-of-life feature when doing things in bulk. Thanks for giving it consideration!

**Reply by luckedea** [reply]
Sounds good, I'll convert it into an issue after some research.

**Reply by bacon-cheeseburger** [reply]
Awesome, thank you!

**Comment by coffinbearer** [comment]
@bacon-cheeseburger i believe you are looking for https://github.com/Prinz23/nzbgetpp

**Reply by ceedveed** [reply]
> @bacon-cheeseburger i believe you are looking for https://github.com/Prinz23/nzbgetpp

Thanks. I already have this extension. It does not work with .7z files.

Is there a solution for .7z files yet?

**Reply by bacon-cheeseburger** [reply]
> > @bacon-cheeseburger i believe you are looking for https://github.com/Prinz23/nzbgetpp
> 
> Thanks. I already have this extension. It does not work with .7z files.
> 
> Is there a solution for .7z files yet?

Maybe you're using an outdated version because it does support 7z. 

https://github.com/Prinz23/nzbgetpp/blob/master/unzip.py#L328-L341

**Reply by ceedveed** [reply]
Thanks for pointing this out. I already have the latest version. I have to troubleshoot further. I think it's because I'm running Python 2.x instead of Python 3 on the Windows 10 host.

```
NOTE: This script requires Python to be installed on your system.
#       Optional support of py7zr (https://pypi.org/project/py7zr/)
#       under python 3
```

**Reply by bacon-cheeseburger** [reply]
> Thanks for pointing this out. I already have the latest version. I have to troubleshoot further. I think it's because I'm running Python 2.x instead of Python 3 on the Windows 10 host.

That would be the issue, py7zr requires Python 3. I'd think simply installing it would resolve your problem.

**Reply by ceedveed** [reply]
> > Thanks for pointing this out. I already have the latest version. I have to troubleshoot further. I think it's because I'm running Python 2.x instead of Python 3 on the Windows 10 host.
> 
> That would be the issue, py7zr requires Python 3. I'd think simply installing it would resolve your problem.

I installed Python 3, set PATH, verified python 3 present in windows command prompt, installed py7zr through pip. It still won't work with .7z files.

I tried adding the .7z file to the incoming nzb folder & adding the .7z file using the NZBget interface. Did not work. I probably need to open an issue in the [nzbgetpp](https://github.com/Prinz23/nzbgetpp)

**Reply by bacon-cheeseburger** [reply]
> I installed Python 3, set PATH, verified python 3 present in windows command prompt, installed py7zr through pip. It still won't work with .7z files.
> 
> I tried adding the .7z file to the incoming nzb folder & adding the .7z file using the NZBget interface. Did not work. I probably need to open an issue in the [nzbgetpp](https://github.com/Prinz23/nzbgetpp)

I just tested it here and it worked. For reference I'm using Linux with python3 3.12.7 and py7zr 0.22.0.

**Reply by ceedveed** [reply]
Realized I was still running v21 of NZBget. Thought my issue is related to https://github.com/nzbgetcom/nzbget/pull/8

Updated to v24.5 stable. Enabled the unzip.py extension in the new extension manager. Still won't work with .7z files. Tested with zip files and it works fine. Will continue to troubleshoot.

EDIT: SOLVED - needed to use the ShellOverride setting within NZBGet to set the file path to the python 3 executable in windows. This linked helped me: https://forum.nzbget.net/viewtopic.php?t=2720
